logictest: make local-legacy-schema-changer a default config

fqazi opened the pull request:
The `local-legacy-schema-changer` config was never run in CI because it
was not added to the list of default configs. This commit fixes that.
Some instructions for adding new configs have been added above the list
of configs.

Release note: None

Release justification: low risk and extends test coverage

mgartner committed:
logictest: make local-legacy-schema-changer a default config

Fixes: #87227

The `local-legacy-schema-changer` config was never run in CI because it
was not added to the list of default configs. This commit fixes that.
Some instructions for adding new configs have been added above the list
of configs.

Release note: None
Release justification: low risk extending our test coverage

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/87245)
<!-- Reviewable:end -->

rickystewart commented on pkg/sql/logictest/logictestbase/logictestbase.go:
```diff
@@ -240,13 +240,19 @@ var multiregion15node5region3azsLocalities = map[int]roachpb.Locality{
-// via -config).
+//
+// If no configs are indicated in a test file, the default configs are used
+// (unless overridden via -config). See DefaultConfigNames for the list of
```
Drive by: Note, the `-config` flag doesn't exist any more.

fqazi committed:
sql: drop sequence does not clean all dependencies

Previously, when dropping a sequence with the cascade
option on the legacy schema changer, we accidentally
regressed this adding cascaded drops for UDFs. This
patch fixes an early return that prevented us from
dropping more then one dependent view.

Release justification: low risk and fixes a regression
in 22.2
Release note: None

fqazi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87245)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rickystewart)

___
*[`pkg/sql/logictest/logictestbase/logictestbase.go` line 250 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/87245#-NArHxbmbOxxtZY_OfcD-r1-250:-NAtu3_a-l_JzKxmLhzH:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/e74c8c4e35cab63b1c4675824b024ba8a99ed97b/pkg/sql/logictest/logictestbase/logictestbase.go#L250)):*
<details><summary><i>Previously, rickystewart (Ricky Stewart) wrote…</i></summary><blockquote>

Drive by: Note, the `-config` flag doesn't exist any more.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

ajwerner commented on pkg/sql/logictest/testdata/logic_test/fk:
```diff
@@ -4137,6 +4137,9 @@ SET CLUSTER SETTING jobs.registry.interval.base = .00001;
 statement ok
 SET CLUSTER SETTING jobs.debug.pausepoints = 'schemachanger.before.exec'
 
+# See issue #87266, this table prevents the legacy schema changer from dropping
+# the database, which is unexpected
+skipif config local-legacy-schema-changer
```
is this now fixed?

ajwerner reviewed (approved):
LGTM mod the one skip I think you fixed.

fqazi reviewed (commented):
That one still needs to be investigated, I'll spend time on it a in a bit.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87245#-:-NAu9Zxm7xRGHVvHu2_t:b6s9pi7)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @rickystewart)

___
*[`pkg/sql/logictest/testdata/logic_test/fk` line 4142 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/87245#-NArHxbmbOxxtZY_OfcH-r3-4142:-NAu9c4yA8N7k5SxrrAO:b5yqxxl) ([raw file](https://github.com/cockroachdb/cockroach/blob/9db4479f088af6e528b1742931c4b99448569709/pkg/sql/logictest/testdata/logic_test/fk#L4142)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

is this now fixed?
</blockquote></details>

Unfortunately, this one is a different issue. I'll try to root cause it today


<!-- Sent from Reviewable.io -->

fqazi commented:
TFTR @rickystewart @ajwerner !

bors r+

ajwerner commented:
bors r-

gotta generate

ajwerner commented:
after a rebase

fqazi commented:
bors r+
